cmd/createorupdate/createorupdate.go

```diff
@@ -441,6 +449,146 @@ func createResourceGroup(log *logrus.Entry) error {
 	return err
 }
 
+func gatherArtifacts(artifactDir, artifactKubeconfig string) error {
```

Comment:
Does it make sense to move these to a separate file as createorupdate.go is becoming kinda big?

Reply:
We will have to add the new file in the `go run` invocation, and I have been resisting it so far but I definitely agree this file should be split into separate self-contained files. Certainly the server code deserves its own `server.go` place.

Reply:
Will do the split in a follow-up